iroh-embed/src/lib.rs

```diff
@@ -0,0 +1,259 @@
+//! Using iroh's peer-to-peer IPFS as a library.
+//!
+//! This crate supports embedding iroh in another application, enables running an IPFS node.
+//! By default the system will not interfere with any other IPFS node, iroh or otherwise, on
+//! the same host.
+//!
+//! An iroh system consists of several services, depending on how the IPFS node should
+//! behave not all of them may be needed:
+//!
+//! - The **store** service.  Iroh needs somewhere to store data, currently only an on-disk
+//!   storage is available as [`RocksStoreService`], in the future an in-memory version will
+//!   be available too.
+//! - The **p2p** service.  This service communicates with the wider IPFS network.
+//! - The **gateway** service.  This provides an HTTP gateway into IPFS.  Currently not yet
+//!   supported in iroh-embed.
+//!
+//! The current work-in-progress version of iroh-embed only allows using both a store and
+//! p2p service.  Future combinations will become available as features are added.
+//!
+//! # Getting started
+//!
+//! To create an iroh system you will need a few things:
+//!
+//! - Create a store service using [`RocksStoreService`].
+//! - Create a p2p service using [`P2pService`] and hooking it up to your earlier created
+//!   store service.
+//! - Create the [`Iroh`] system using [`IrohBuilder`].
+//! - Use the system using [`Iroh::api`].
+//!
+//! An example is available in the repository under `examples/embed`.
+
+use anyhow::{bail, Result};
+use iroh_rpc_client::Config as RpcClientConfig;
+
+pub use iroh_api::Api;
+pub use iroh_p2p::Libp2pConfig;
+pub use iroh_unixfs::indexer::IndexerUrl;
+pub use reqwest::Url;
+
+/// An address allowing internal communication with the iroh store service.
+///
+/// This is only needed to wire up the several services together and should not be used to
+/// communicate directly to the service.
+///
+/// Can be created from [`store::RocksStoreService::addr`].
```

Comment:
seems you mixed this up, this is the p2p addr and below you refer to p2p on the store addr

Reply:
@flub I think you missed this one